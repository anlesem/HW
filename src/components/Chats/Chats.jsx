
import style from './Chats.module.scss';

import List from '@mui/material/List';
import ListItem from '@mui/material/ListItem';
import ListItemText from '@mui/material/ListItemText';
import IconButton from '@mui/material/IconButton';
import CommentIcon from '@mui/icons-material/Comment';

export const Chats = (props) => {
   const users = props.message.filter(item => item.author !== 'BOT').reduce((acc, cur) => {
      const same = acc.find((element) => element.author === cur.author);
      if (!same) acc.push(cur);
      return acc;
   }, []);

   return (
      <div className={style.chats}>Чаты:
         <List sx={{ width: '100%', maxWidth: 360, bgcolor: 'background.paper' }}>
            {users.map((item) => (
               <ListItem
                  key={item.id}
                  disableGutters
                  secondaryAction={
                     <IconButton>
                        <CommentIcon />
                     </IconButton>
                  }
               >
                  <ListItemText primary={`Пользователь ${item.author}`} />
               </ListItem>
            ))}
         </List>
      </div>
   );
}