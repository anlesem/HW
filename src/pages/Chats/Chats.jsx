import { useState, useEffect, useCallback } from 'react';
import { useParams, useNavigate } from 'react-router-dom';

import { ChatList } from '../../components/ChatList/ChatList';
import { Message } from '../../components/Message/Message';

import style from './Chats.module.scss';

const defaultChats = [
   {
      id: '1',
      name: 'чат 1',
   },
];

export const Chats = () => {
   const [chats, setChats] = useState(defaultChats);
   const [counterId, setCounterId] = useState(chats.length);
   const [chatIs, setChatIs] = useState(false);
   const { chatId } = useParams();
   const navigate = useNavigate();

   const addChat = useCallback(() => {
      setChats((prevChats) => ([
         ...prevChats, {
            id: `${counterId + 1}`,
            name: `чат ${counterId + 1}`,
         },
      ]));
      setCounterId(counterId + 1);
   }, [counterId])

   const renameChat = useCallback((id, name) => {
      let update = [...chats];

      update.find(item => item.id === `${id}`).name = name;
      setChats(update);
   }, [chats])

   const removeChat = useCallback((checked) => {
      let update = [...chats];
      checked.forEach(id => {
         update = update.filter(item => item.id !== `${id}`);
      });
      setChats(update);
   }, [chats])

   useEffect(() => {
      if (chats.length === 0) addChat();
   }, [addChat, chats, removeChat]);

   useEffect(() => {
      if (chats.findIndex(item => item.id === `${chatId}`) < 0) {
         navigate("/chats");
      } else setChatIs(true);
   }, [chatId, chats, navigate]);

   return (
      <div className={style.main}>
         <div className={style.chats}>Чаты:
            <ChatList chatData={{ chatId, chats, addChat, removeChat, renameChat }} />
         </div>
         <div className={style.messages} >Сообщения:
            {chatId && chatIs ? <Message chatId={chatId} /> : <Message chatId="0" />}
         </div>
      </div>
   );
}