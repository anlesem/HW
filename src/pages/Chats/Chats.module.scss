.chats {
   width: 30%;
   margin-top: 40px;
   padding-left: 20px;
   font-size: 20px;
   font-weight: semibold;
}