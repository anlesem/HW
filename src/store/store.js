import { createStore, combineReducers, compose, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { persistStore, persistReducer } from 'redux-persist';
import storage from 'redux-persist/lib/storage';

import { profileReducer } from './profile/reducer';
import { chatsReducer } from './chats/reducer';
import { messagesReducer } from './messages/reducer';
import { dataReducer } from './data/reducer';

const persistConfig = {
  key: 'root',
  storage,
  blacklist: ['data']
};

const rootReducers = combineReducers({
  profile: profileReducer,
  chats: chatsReducer,
  messages: messagesReducer,
  data: dataReducer
});

const persistedReducers = persistReducer(persistConfig, rootReducers);

export const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export const store = createStore(persistedReducers, composeEnhancers(applyMiddleware(thunk)));

export const persistor = persistStore(store);
