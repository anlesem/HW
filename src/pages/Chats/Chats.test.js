import React from 'react';
import { BrowserRouter, Navigate, Route, Routes } from 'react-router-dom';
import { render, screen, fireEvent } from '@testing-library/react';
import { Chats } from './Chats';

import { Provider } from 'react-redux';
import { store } from '../../store/store';

describe('Chats', () => {
   it('Компонент существует', () => {
      expect(Chats).toBeInstanceOf(Function);
   });

   it('Состояние при переходе на вкладку Чаты', () => {
      render(<Provider store={store}><BrowserRouter>
         <Routes>
            <Route path="chats" element={
               <React.Suspense fallback={<>...</>}>
                  <Chats />
               </React.Suspense>} />
            <Route path="*" element={<Navigate to="chats" />} />
         </Routes>
      </BrowserRouter></Provider>);

      const buttonRemove = screen.getByTestId('chats-button-remove');
      const inputMessage = screen.getByTestId('message-form-input');
      const buttonMessage = screen.getByTestId('message-form-button');

      expect(screen.getByText(/чат 1/i)).toBeTruthy();
      expect(screen.getByText(/выберите чат/i)).toBeTruthy();
      expect(screen.getByTestId('chat-item-1')).toBeTruthy();
      expect(screen.queryByTestId('chat-item-2')).toBeFalsy();

      expect(buttonRemove).toHaveClass("Mui-disabled");
      expect(inputMessage).toHaveClass("Mui-disabled");
      expect(buttonMessage).toHaveClass("Mui-disabled");
   });

   it('Добавление чата', () => {
      render(<Provider store={store}><BrowserRouter>
         <Routes>
            <Route path="chats" element={
               <React.Suspense fallback={<>...</>}>
                  <Chats />
               </React.Suspense>} />
            <Route path="*" element={<Navigate to="chats" />} />
         </Routes>
      </BrowserRouter></Provider>);

      const buttonAdd = screen.getByTestId('chats-button-add');

      fireEvent.click(buttonAdd);
      expect(screen.getByTestId('chat-item-1')).toBeTruthy();
      expect(screen.getByTestId('chat-item-2')).toBeTruthy();
      expect(screen.queryByTestId('chat-item-3')).toBeFalsy();

      fireEvent.click(buttonAdd);
      expect(screen.getByTestId('chat-item-3')).toBeTruthy();
      expect(screen.queryByTestId('chat-item-4')).toBeFalsy();
   });

   it('Удаление чата', () => {
      render(<Provider store={store}><BrowserRouter>
         <Routes>
            <Route path="chats" element={
               <React.Suspense fallback={<>...</>}>
                  <Chats />
               </React.Suspense>} />
            <Route path="*" element={<Navigate to="chats" />} />
         </Routes>
      </BrowserRouter></Provider>);

      const chatItem1 = screen.getByTestId('chat-item-1');
      const buttonAdd = screen.getByTestId('chats-button-add');
      const buttonRemove = screen.getByTestId('chats-button-remove');
      const chatInput1 = screen.getByTestId('chat-check-1');

      expect(chatInput1.checked).toBeFalsy();
      expect(screen.getByText(/чат 1/i)).toBeTruthy();
      expect(buttonRemove).toHaveClass("Mui-disabled");
      expect(screen.queryByTestId('chat-form-input')).toBeFalsy();
      expect(screen.queryByTestId('chat-form-button')).toBeFalsy();


      fireEvent.click(chatItem1);
      expect(chatInput1.checked).toBeTruthy();
      expect(buttonRemove).not.toHaveClass("Mui-disabled");
      expect(screen.getByTestId('chat-form-input')).toBeTruthy();
      expect(screen.getByTestId('chat-form-button')).toBeTruthy();

      fireEvent.click(buttonRemove);
      expect(screen.getByText(/чат 2/i)).toBeTruthy();
      expect(buttonRemove).toHaveClass("Mui-disabled");
      expect(screen.queryByTestId('chat-item-1')).toBeFalsy();
      expect(screen.getByTestId('chat-item-2')).toBeTruthy();

      fireEvent.click(buttonAdd);
      fireEvent.click(buttonAdd);
      expect(screen.getByText(/чат 3/i)).toBeTruthy();
      expect(screen.getByText(/чат 4/i)).toBeTruthy();
      expect(buttonRemove).toHaveClass("Mui-disabled");
      expect(screen.queryByTestId('chat-form-input')).toBeFalsy();
      expect(screen.queryByTestId('chat-form-button')).toBeFalsy();

      const chatItem2 = screen.getByTestId('chat-item-2');
      const chatItem3 = screen.getByTestId('chat-item-3');
      const chatItem4 = screen.getByTestId('chat-item-4');
      const chatInput2 = screen.getByTestId('chat-check-2');
      const chatInput3 = screen.getByTestId('chat-check-3');
      const chatInput4 = screen.getByTestId('chat-check-4');

      fireEvent.click(chatItem2);
      expect(chatInput2.checked).toBeTruthy();
      expect(buttonRemove).not.toHaveClass("Mui-disabled");
      expect(screen.getByTestId('chat-form-input')).toBeTruthy();
      expect(screen.getByTestId('chat-form-button')).toBeTruthy();

      fireEvent.click(chatItem3);
      fireEvent.click(chatItem4);
      expect(chatInput3.checked).toBeTruthy();
      expect(chatInput4.checked).toBeTruthy();
      expect(buttonRemove).not.toHaveClass("Mui-disabled");
      expect(screen.queryByTestId('chat-form-input')).toBeFalsy();
      expect(screen.queryByTestId('chat-form-button')).toBeFalsy();

      fireEvent.click(buttonRemove);
      expect(screen.getByText(/чат 5/i)).toBeTruthy();
      expect(buttonRemove).toHaveClass("Mui-disabled");
      expect(screen.queryByTestId('chat-item-4')).toBeFalsy();
      expect(screen.getByTestId('chat-item-5')).toBeTruthy();
      expect(screen.queryByTestId('chat-item-6')).toBeFalsy();
      expect(screen.queryByTestId('chat-form-input')).toBeFalsy();
      expect(screen.queryByTestId('chat-form-button')).toBeFalsy();
   });

   it('Переименование чата', () => {
      render(<Provider store={store}><BrowserRouter>
         <Routes>
            <Route path="chats" element={
               <React.Suspense fallback={<>...</>}>
                  <Chats />
               </React.Suspense>} />
            <Route path="*" element={<Navigate to="chats" />} />
         </Routes>
      </BrowserRouter></Provider>);

      const buttonAdd = screen.getByTestId('chats-button-add');
      const chatItem1 = screen.getByTestId('chat-item-1');

      expect(screen.getByText(/чат 1/i)).toBeTruthy();

      fireEvent.click(chatItem1);

      const inputName = screen.getByTestId('chat-form-input');
      const buttonName = screen.getByTestId('chat-form-button');

      fireEvent.change(inputName, { target: { value: 'Вася' } });
      expect(inputName.value).toBe('Вася');

      fireEvent.click(buttonName);
      expect(screen.getByText(/вася/i)).toBeTruthy();
      expect(screen.queryByTestId('chat-form-input')).toBeFalsy();
      expect(screen.queryByTestId('chat-form-button')).toBeFalsy();

      fireEvent.click(buttonAdd);
      expect(screen.getByText(/чат 2/i)).toBeTruthy();

      const chatItem2 = screen.getByTestId('chat-item-2');

      fireEvent.click(chatItem2);
      fireEvent.change(inputName, { target: { value: 'Петя' } });

      fireEvent.click(chatItem1);
      expect(screen.queryByTestId('chat-form-input')).toBeFalsy();
      expect(screen.queryByTestId('chat-form-button')).toBeFalsy();

      fireEvent.click(chatItem1);
      fireEvent.click(buttonName);
      expect(screen.getByText(/вася/i)).toBeTruthy();
      expect(screen.getByText(/чат 2/i)).toBeTruthy();
   });

   it('Переименование чата по нажатию Enter', () => {
      render(<Provider store={store}><BrowserRouter>
         <Routes>
            <Route path="chats" element={
               <React.Suspense fallback={<>...</>}>
                  <Chats />
               </React.Suspense>} />
            <Route path="*" element={<Navigate to="chats" />} />
         </Routes>
      </BrowserRouter></Provider>);

      const chatItem1 = screen.getByTestId('chat-item-1');
      expect(screen.getByText(/вася/i)).toBeTruthy();

      fireEvent.click(chatItem1);

      const inputName = screen.getByTestId('chat-form-input');

      fireEvent.change(inputName, { target: { value: 'Петя' } });
      expect(inputName.value).toBe('Петя');

      fireEvent.keyDown(inputName, { key: 'enter', keyCode: 13 });
      expect(screen.getByText(/петя/i)).toBeTruthy();
      expect(screen.queryByTestId('chat-form-input')).toBeFalsy();
      expect(screen.queryByTestId('chat-form-button')).toBeFalsy();
   });
});

describe('Chats/id', () => {
   it('Состояние при открытии чата', () => {
      render(<Provider store={store}><BrowserRouter>
         <Routes>
            <Route path="chats/:chatId" element={
               <React.Suspense fallback={<>...</>}>
                  <Chats />
               </React.Suspense>} />
            <Route path="*" element={<Navigate to="chats/1" />} />
         </Routes>
      </BrowserRouter></Provider>);

      const inputMessage = screen.getByTestId('message-form-input');
      const buttonMessage = screen.getByTestId('message-form-button');

      expect(screen.getByText(/Добро пожаловать в чат №1/)).toBeTruthy();
      expect(screen.getByTestId('list-item-0')).toBeTruthy();
      expect(inputMessage).not.toHaveClass("Mui-disabled");
      expect(buttonMessage).not.toHaveClass("Mui-disabled");
   });

   it('Добавление сообщения', () => {
      render(<Provider store={store}><BrowserRouter>
         <Routes>
            <Route path="chats/:chatId" element={
               <React.Suspense fallback={<>...</>}>
                  <Chats />
               </React.Suspense>} />
            <Route path="*" element={<Navigate to="chats/1" />} />
         </Routes>
      </BrowserRouter></Provider>);

      const inputMessage = screen.getByTestId('message-form-input');
      const buttonMessage = screen.getByTestId('message-form-button');

      fireEvent.change(inputMessage, { target: { value: 'SomeMessage' } });
      expect(inputMessage.value).toBe('SomeMessage');
      fireEvent.click(buttonMessage);
      expect(screen.getByText(/SomeMessage/)).toBeTruthy();
      expect(screen.getByTestId('list-item-1')).toBeTruthy();
      expect(screen.queryByTestId('list-item-2')).toBeFalsy();

      setTimeout(() => {
         expect(screen.getByTestId('list-item-2')).toBeTruthy();

         fireEvent.change(inputMessage, { target: { value: 'AnotherMessage' } });
         expect(inputMessage.value).toBe('AnotherMessage');

         fireEvent.keyDown(inputMessage, { key: 'enter', keyCode: 13 })
         expect(screen.getByText(/AnotherMessage/)).toBeTruthy();
         expect(screen.getByTestId('list-item-3')).toBeTruthy();
         expect(screen.queryByTestId('list-item-4')).toBeFalsy();
      }, 2000);
   });

   it('Изменение значения поля ввода при переключении между чатами', () => {
      render(<Provider store={store}><BrowserRouter>
         <Routes>
            <Route path="chats/:chatId" element={
               <React.Suspense fallback={<>...</>}>
                  <Chats />
               </React.Suspense>} />
            <Route path="*" element={<Navigate to="chats/1" />} />
         </Routes>
      </BrowserRouter></Provider>);

      const buttonAdd = screen.getByTestId('chats-button-add');
      const inputMessage = screen.getByTestId('message-form-input');

      fireEvent.click(buttonAdd);
      expect(screen.getByTestId('chat-item-1')).toBeTruthy();
      expect(screen.getByTestId('chat-item-2')).toBeTruthy();

      const chatChange1 = screen.getByTestId('chat-change-1');
      const chatChange2 = screen.getByTestId('chat-change-2');

      fireEvent.change(inputMessage, { target: { value: 'SomeMessage' } });
      expect(inputMessage.value).toBe('SomeMessage');

      fireEvent.click(chatChange2);
      expect(screen.getByText(/Добро пожаловать в чат №2/)).toBeTruthy();
      expect(inputMessage.value).toBe('');

      fireEvent.click(chatChange1);
      expect(screen.getByText(/Добро пожаловать в чат №1/)).toBeTruthy();
      expect(inputMessage.value).toBe('SomeMessage');
   });

});