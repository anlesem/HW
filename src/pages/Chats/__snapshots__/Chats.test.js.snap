// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chats Снимок состояния 1`] = `
<DocumentFragment>
  <div
    class="main"
  >
    <div
      class="chats"
    >
      Чаты user:
      <div
        class="wrap"
      >
        <ul
          class="MuiList-root MuiList-padding chatList css-1mcnwpj-MuiList-root"
        >
          <li
            class="MuiListItem-root MuiListItem-gutters css-g28sfz-MuiListItem-root"
          >
            <div
              class="MuiListItemButton-root MuiListItemButton-dense MuiListItemButton-gutters MuiButtonBase-root css-1m4u3pv-MuiButtonBase-root-MuiListItemButton-root"
              data-testid="chat-item-1"
              tabindex="0"
            >
              <div
                class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
              >
                <span
                  class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root PrivateSwitchBase-edgeStart css-zkqepn-MuiButtonBase-root-MuiCheckbox-root"
                >
                  <input
                    aria-labelledby="checkbox-list-label-1"
                    class="PrivateSwitchBase-input css-1m9pwf3"
                    data-indeterminate="false"
                    data-testid="chat-check-1"
                    tabindex="-1"
                    type="checkbox"
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="CheckBoxOutlineBlankIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                    />
                  </svg>
                </span>
              </div>
              <div
                class="MuiListItemText-root MuiListItemText-dense css-tlelie-MuiListItemText-root"
                id="checkbox-list-label-1"
              >
                <span
                  class="MuiTypography-root MuiTypography-body2 MuiListItemText-primary css-et1ao3-MuiTypography-root"
                >
                  чат 1
                </span>
              </div>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </div>
            <div
              class="MuiListItemSecondaryAction-root css-518kzi-MuiListItemSecondaryAction-root"
            >
              <a
                class=""
                href="/chats/1"
              >
                <button
                  aria-label="comments"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                  data-testid="chat-change-1"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="CommentIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M21.99 4c0-1.1-.89-2-1.99-2H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h14l4 4-.01-18zM18 14H6v-2h12v2zm0-3H6V9h12v2zm0-3H6V6h12v2z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </a>
            </div>
          </li>
        </ul>
        <div
          class="manage css-e53awj-MuiStack-root"
        >
          <button
            class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root Mui-disabled  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
            data-testid="chats-button-remove"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="DeleteIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                />
              </svg>
            </span>
            Удалить
          </button>
          <button
            class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
            data-testid="chats-button-add"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="AddIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                />
              </svg>
            </span>
            Добавить
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="messages"
    >
      Сообщения:
      <div
        class="wrap"
      >
        <ul
          class="messages"
        >
          <li
            class="message"
            data-testid="list-item-0"
          >
            <p
              class="message_text"
            >
              Выберите чат для отображения сообщений
            </p>
            <p
              class="message_author"
            >
              BOT
            </p>
          </li>
        </ul>
        <form
          class="form"
        >
          <div
            class="MuiFormControl-root MuiTextField-root textField css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-disabled Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="outlined-multiline-static"
              id="outlined-multiline-static-label"
            >
              Текст сообщения
              <span
                aria-hidden="true"
                class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
              >
                 *
              </span>
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-formControl MuiInputBase-multiline css-dpjnhs-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <textarea
                aria-invalid="false"
                class="MuiOutlinedInput-input MuiInputBase-input Mui-disabled MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
                data-testid="message-form-input"
                disabled=""
                id="outlined-multiline-static"
                name="text"
                required=""
                rows="2"
                style="height: 0px; overflow: hidden;"
              />
              <textarea
                aria-hidden="true"
                class="MuiOutlinedInput-input MuiInputBase-input Mui-disabled MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
                readonly=""
                style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding: 0px; width: 100%;"
                tabindex="-1"
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-173wfxe"
                >
                  <span>
                    Текст сообщения *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <button
            class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root Mui-disabled  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
            data-testid="message-form-button"
            disabled=""
            tabindex="-1"
            type="submit"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="SendIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M2.01 21 23 12 2.01 3 2 10l15 2-15 2z"
              />
            </svg>
          </button>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
